inference-engine/src/mkldnn_plugin/mkldnn_node.h

```diff
@@ -741,7 +741,8 @@ class MKLDNNNode {
 
     bool inputShapesModified() const;
     virtual bool needShapeInfer() const;
-    std::vector<VectorDims> shapeInferGeneric(const std::vector<Shape>& inputDims) const;
+    std::vector<VectorDims> shapeInferGeneric(const std::vector<Shape>& inputDims = {}, uint32_t value_port_mask = 0) const;
```

Comment:
Also it would be great to have some syntax sugar to perform the following call `shapeInferGeneric(shapes, {1, 3})` where from the `{1, 3}` initialization list 0x1010 bitmask is made.

Reply:
From performance perspective, initialization list introduces run-time cost, can we use alternative syntax like: `shapeInferGeneric(shapes, PortSetMask(1, 3))` ? so we can rely on `constexpr` to keep the cost induced by syntax sugar to compile time: as confirmed in following code: 

https://godbolt.org/z/PhrWb4x58

Reply:
Good solution. May be it makes sense to make the name shorter. What about using just `PortMask`?

Reply:
OK, PortMask added!